server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -924,6 +1015,135 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
+  }
+
+  // Instead of splitting or chopping tablets for a delete we instead create ranges
+  // to exclude the portion of the tablet that should be deleted
+  private List<Range> createRangesForDeletion(TabletMetadata tabletMetadata,
+      final KeyExtent deleteRange) {
+    final KeyExtent keyExtent = tabletMetadata.getExtent();
+    final Text prevTabletEndRow =
+        tabletMetadata.getPrevEndRow() != null ? tabletMetadata.getPrevEndRow() : null;
+    final Text tabletEndRow = keyExtent.endRow();
+    final Range tabletRange = tabletMetadata.getExtent().toDataRange();
+    final Key firstKey =
+        prevTabletEndRow != null ? new Key(prevTabletEndRow).followingKey(PartialKey.ROW) : null;
+    final List<Range> ranges = new ArrayList<>();
+
+    // This covers the case of when a deletion range overlaps the first tablet
+    // We need to create a range that excludes the deletion
+    if ((firstKey == null || !firstKey.equals(deleteRange.toDataRange().getStartKey()))
+        && deleteRange.toDataRange().getStartKey() != null
+        && tabletRange.contains(deleteRange.toDataRange().getStartKey())) {
+      Manager.log.trace(
+          "Fencing tablet at start of deletion range with end row {}, Start deletion row {}",
+          prevTabletEndRow, deleteRange.toDataRange().getStartKey().getRow());
+      ranges.add(new Range(prevTabletEndRow, false,
+          deleteRange.toDataRange().getStartKey().getRow(), false));
+    }
+
+    // This covers the case of when a deletion range overlaps the last tablet
+    // We need to create a range that excludes the deletion.
+    if ((tabletEndRow == null
+        || !tabletEndRow.equals(deleteRange.toDataRange().getEndKey().getRow()))
+        && deleteRange.toDataRange().getEndKey() != null
+        && tabletRange.contains(deleteRange.toDataRange().getEndKey())) {
+      Manager.log.trace(
+          "Fencing tablet at end of deletion range with end row {}, End deletion row {}",
+          tabletEndRow, deleteRange.toDataRange().getEndKey().getRow());
+      ranges.add(new Range(deleteRange.toDataRange().getEndKey().getRow(), tabletEndRow));
+    }
+
+    return ranges;
+  }
+
+  private void updateMetadataRecordsForDelete(MergeInfo info) throws AccumuloException {
+    final KeyExtent range = info.getExtent();
+
+    String targetSystemTable = MetadataTable.NAME;
+    if (range.isMeta()) {
+      targetSystemTable = RootTable.NAME;
+    }
+
+    final AccumuloClient client = manager.getContext();
+
+    try (BatchWriter bw = client.createBatchWriter(targetSystemTable)) {
+      final Text startRow = range.prevEndRow();
+      final Text lastRow = range.endRow() != null
+          ? new Key(range.endRow()).followingKey(PartialKey.ROW).getRow() : null;
+
+      // Find the tablets that overlap the start and end of the deletion range
+      final List<TabletMetadata> tabletMetadatas = new ArrayList<>();
+      loadTabletMetadata(range.tableId(), startRow, ColumnType.PREV_ROW, ColumnType.FILES)
```

Comment:
For this what is the case where the tablet is not present?

Reply:
I don't think there is a case where it wouldn't find a tablet. the loadTabletMetadata() method calls off to Ample and uses the overlapp() method so it should always map and return a matching tablet.